src/Pact/Types/Command.hs

```diff
@@ -179,41 +206,58 @@ data ParsedCode = ParsedCode
 instance NFData ParsedCode
 
 
+
+-- | Signer combines PPKScheme, PublicKey, and the Address (aka the
+--   formatted PublicKey).
+data Signer = Signer
+ { _siScheme :: !PPKScheme
+ , _siPubKey :: !Text
+ , _siAddress :: !Text
+ } deriving (Eq, Ord, Show, Generic)
+
+instance NFData Signer
+instance Serialize Signer
+instance ToJSON Signer where
+  toJSON Signer{..} = object [
+    "scheme" .= _siScheme,
+    "pubKey" .= _siPubKey,
+    "addr" .= _siAddress]
+instance FromJSON Signer where
+  parseJSON = withObject "Signer" $ \o -> do
+    pub <- o .: "pubKey"
+    scheme <- o .:? "scheme"   -- defaults to PPKScheme default
+    addr <- o .:? "addr"       -- defaults to full Public Key
+
+    return $ Signer
+             (fromMaybe defPPKScheme scheme)
+             pub
+             (fromMaybe pub addr)
+
+
+
 -- | Payload combines a 'PactRPC' with a nonce and platform-specific metadata.
 data Payload m c = Payload
   { _pPayload :: !(PactRPC c)
   , _pNonce :: !Text
   , _pMeta :: !m
+  , _pSigners :: ![Signer]
   } deriving (Show, Eq, Generic, Functor, Foldable, Traversable)
 instance (NFData a,NFData m) => NFData (Payload m a)
 instance (ToJSON a,ToJSON m) => ToJSON (Payload m a) where toJSON = lensyToJSON 2
 instance (FromJSON a,FromJSON m) => FromJSON (Payload m a) where parseJSON = lensyParseJSON 2
 
 
--- | UserSig combines PPKScheme, PublicKey, and the formatted PublicKey
---   referred to as the Address.
-data UserSig = UserSig
-  { _usScheme :: !PPKScheme
-  , _usPubKey :: !Text
-  , _usAddress :: !Text
-  , _usSig :: !Text }
-  deriving (Eq, Ord, Show, Generic)
-instance NFData UserSig
 
+data UserSig = UserSig { _usSig :: !Text }
+  deriving (Eq, Ord, Show, Generic)
```

Comment:
can we `newtype` this?

Reply:
When I make this a newtype, I get 
```A newtype constructor cannot have a strictness annotation```

Reply:
Newtypes don't need strictness, this is a good change

Reply:
See latest commit